File: backend/app/schemas/telemetry.py
from pydantic import BaseModel

class DriverState(BaseModel):
    id: str
    driverNumber: int
    name: str
    team: str
    position: int
    lap: int
    trackPercentage: float
    gapToLeader: str
    lastLapTime: str
    bestLapTime: str
    sector1: str
    sector2: str
    sector3: str
    tyreCompound: str
    tyreAge: int
    pitStops: int
    inPit: bool

File: frontend/src/types/telemetry.ts
export interface DriverState {
  id: string;
  driverNumber: number;
  name: string;
  team: string;
  position: number;
  lap: number;
  trackPercentage: number;
  gapToLeader: string;
  lastLapTime: string;
  bestLapTime: string;
  sector1: string;
  sector2: string;
  sector3: string;
  tyreCompound: string;
  tyreAge: number;
  pitStops: number;
  inPit: boolean;
}

export interface TelemetryUpdate {
  event: string;
  data: DriverState[];
}